docs/api-reference.md

```diff
@@ -219,6 +219,126 @@ Response:
+
+* `frame` (string): The Frame that will contain this action's set bits.
+* `rowid` (int): The action can use this as a pre-defined SetBit rowID.  The user is required to ensure this ID does not overlap with other rows in use per frame.
+* `valueDestination` (string): The mapping rule used for this data.
+    - `value-to-row`: The value should be an integer and will map directly to a RowID.
+    - `single-row-boolean`: If the value is true set a bit using the `rowid`.
+    - `mapping`: Map the value to a RowID in the `valueMap`.
+* `valueMap` (object): string and integer pairs used to map field values to RowID's.
+ 
+Request:
+```
+curl localhost:10101/index/user/input-definition/stargazer-input \
+    -X POST \
+    -d '{
+            "frames":[
+                {
+                    "name": "language",
+                    "options": {"rowLabel": "language_id"}
+                }
+            ],
+            "fields":[
+                {
+                    "name": "repo_id",
+                    "primaryKey":true
+                },
+                {
+                    "name": "language_id",
+                    "actions":[
+                        {
+                            "frame": "language",
+                            "valueDestination": "mapping",
+                            "valueMap": {
+                                "Go": 5,
+                                "Python": 17,
+                                "C++": 10
+                            }
+                        }
+                    ]
+                }
+            ]
+        }'
+```
+ 
+Response:
+```
+{}
+```
+
+### Get input definition
+
+`GET /index/<index-name>/input-definition/<input-definition-name>`
+
+Returns the given input definition as JSON.
+
+Request:
+```
+curl -XGET localhost:10101/index/user/input-definition/stargazer-input
+```
+
+Response:
+```
+{"frames":[{"name":"language","options":{"rowLabel":"language_id"}}],"fields":[{"name":"repo_id","primaryKey":true},{"name":"language_id","actions":[{"frame":"language","valueDestination":"mapping","valueMap":{"Go":5,"Python":17,"C++":10}}]}]}
+```
+
+### Remove input definition
+
+`DELETE /index/<index-name>/input-definition/<input-definition-name>`
+
+Removes the given input definition.
+
+Request:
+```
+curl -XDELETE localhost:10101/index/user/input-definition/stargazer-input
+```
+
+Response:
+```
+{}
+```
+
+### Process input data
+
+`POST /index/<index-name>/input/<input-definition-name>`
+
+Processes the JSON payload using the given input definition.
+
+The request payload is a JSON array of objects containing one field for the primary key that corresponds to the column label, and additional fields that will be handled by corresponding actions in the input definition.
+
+Request:
+```
+curl localhost:10101/index/user/input/stargazer-input \
+     -X POST \
+     -d '[{"language_id": "Go", "repo_id": 92274475, "stargazer_id": 513111}]'
```

Comment:
I think either we remove stargazer_id here or add frame/fields to stargazer in input-definition for a clear example.